### ISSUE TITLE
Incorrectly warns for variables in list comprehensions with {hoist_local: false}

### ISSUE THREAD

**Issue by saifelse** [za-creature/coffeescope#16]
``` zsh
echo '(x for x in [1,2,3])' | coffeelint -s -f =(echo '
{
     "check_scope": {
        "module": "coffeescope2",
        "same_scope": false,
        "hoist_local": false
    }
}
')
```

outputs 

```
  ⚡ stdin
     ⚡ #1: Variable "x" used before it was first defined (on line 1, column 8).
```

**Comment by za-creature** [comment]
Will give this a go, but it's nontrivial because:

``` coffee
(x for x in [1,2,3])
```

has the same AST as

``` coffee
(
    for x in [1, 2, 3]
        x
)
```

**Comment by za-creature** [comment]
A bit hacky, but fixed in 0.3.2. Please reopen if this doesn't solve your use case.

**Comment by saifelse** [comment]
Thanks for your expediency:

```
(x for x in [1, 2, 3])
(x for x in [1, 2, 3])
```

also errors with:

```
     ⚡ #1: Variable "x" used before it was first defined (on line 1, column 8).
```

**Comment by za-creature** [comment]
Sorry for taking a while to get to this. After giving the matter a lot of thought, I believe I will revert my previous commit and always hoist in comprehensions, thus missing some edge cases like
(x+y for x in [1..10] for y in [1..x])

Reason behind this is that coffeescript's scoping rules are weird at times and both the AST and the generated code in `for` loops contain a generous amount of insanity, as the author(s) themselves state: http://coffeescript.org/documentation/docs/nodes.html#section-160

**Comment by za-creature** [comment]
Fix landed in 0.3.5, however you should expect false negatives when using comprehensions
